src/config/n_plus_one.rs

```diff
@@ -20,7 +20,7 @@ pub fn n_plus_one(config: &Config) -> Vec<Vec<(String, String)>> {
           } else if field.has_resolver() && !field.has_batched_resolver() && is_list {
             vec![new_path]
           } else {
-            find_fan_out(config, &field.type_of, new_path, field.list.is_some() || is_list)
+            find_fan_out(config, &field.type_of, new_path, field.list.unwrap_or(false) || is_list)
```

Comment:
As for example from #342 the User gets `.list = Some(false)` and that was wrongly handled by n+1 check code.

I've a little bit confused of using Option<bool> for such kind of flag instead of pure bool, but there are several usages of it and I don't want to break anything in case it is important

Reply:
Config can read directly from a .json or a yaml file. That allows A user to directly write Config and we build a blueprint directly from it. Option of bool is used so that the user doesn't need to specify every boolean option and we can assume certain defaults.

Reply:
Hope this makes sense, happy to answer more questions if you have any.

Reply:
Thanks for the explanation, then current changes should be ok to fix the issue.

Speaking of reading from config - using [serde's default](https://serde.rs/attr-default.html) might be helpful for this. But it is out of scope for the current issue

Reply:
@meskill Thanks! Would you mind creating an issue for it?

Reply:
Created https://github.com/tailcallhq/tailcall/issues/367